### ISSUE TITLE
re-enable package will cause duplicate lint error report or more

### ISSUE THREAD

**Issue by noe132** [AtomLinter/linter-eslint#1153]
## Issue Type
Bug

## Issue Description
1. Open atom with new profile
2. install `linter linter-eslint linter-ui-default bust-signal intensions`
3. init a empty folder, and install eslint
4. create `.eslintrc.js` with following content
```js
module.exports = {
  'rules': {
    'semi': ['error', 'never'],
  },
}
```
5. create a index.js and type in `console.log('hi');`
from now you will get lint result about unnecessary semicolon
6. go to setting page, disable `linter-eslint` and re-enable it
7. you might check that extra semi was reported twice
8. repeat 6 and 7, and you might get several extra semi report after re-enable the package
![tim 20180724173243](https://user-images.githubusercontent.com/2271900/43129696-9707a26a-8f67-11e8-8e15-622910c8b092.png)

it looks like the previous eslint instance wasn't get closed, and re-enable the package creates a new one that they both provide linting message
```
Atom version: 1.28.2
linter-eslint version: 8.4.1
ESLint version: 5.2.0
Hours since last Atom restart: 0.1
Platform: win32
Using local project ESLint from: C:\Users\noe\src\etest\node_modules\eslint
Current file's scopes: [
  "source.js",
  "punctuation.terminator.statement.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDirs": [],
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by skylize** [comment]
@Arcanemagus Seems like we should force clear any existing messages as part of `deactivate`.
